astra-sim/workload/Workload.cc

```diff
@@ -103,45 +102,53 @@ void Workload::issue_dep_free_nodes() {
 
   while (!push_back_queue.empty()) {
     shared_ptr<Chakra::ETFeederNode> node = push_back_queue.front();
-    et_feeder->pushBackIssuableNode(node->getChakraNode()->id());
+    et_feeder->pushBackIssuableNode(node->id());
     push_back_queue.pop();
   }
 }
 
 void Workload::issue(shared_ptr<Chakra::ETFeederNode> node) {
-  if ((node->getChakraNode()->node_type() == ChakraNodeType::MEM_LOAD_NODE) ||
-      (node->getChakraNode()->node_type() == ChakraNodeType::MEM_STORE_NODE)) {
-    if (sys->trace_enabled) {
-      cout << "issue,sys->id=" << sys->id << ",tick=" << Sys::boostedTick()
-           << ",node->id=" << node->getChakraNode()->id()
-           << ",node->name=" << node->getChakraNode()->name() << endl;
-    }
-    issue_remote_mem(node);
-  } else if (node->getChakraNode()->node_type() == ChakraNodeType::COMP_NODE) {
-    if ((node->getChakraNode()->simulated_run_time() == 0) &&
-        (node->getChakraNode()->num_ops() == 0)) {
-      skip_invalid(node);
-    } else {
+  if (sys->replay_only) {
+      issue_comp(node);
```

Comment:
If replay_only means we are relying on the recorded execution times for all nodes, including communication nodes, then it is confusing here to call "issue_comp" on all nodes.  We should rename "issue_comp" to something to the effect "advance_time" or "use_recorded_latency".  Also within that newly renamed function, we should assert that the passed in node has a recorded time.

Reply:
It has been replaced with issue_replay.